substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -687,47 +705,119 @@ impl<T: Config> Pallet<T> {
 					CurrentEra::<T>::put(0);
 					ErasStartSessionIndex::<T>::insert(&0, &start_session_index);
 				},
-				_ => (),
+				_ => {},
 			}
 
+			// election failed, clear election prep metadata.
+			Self::clear_election_metadata();
```

Comment:
`clear_election_metadata` is called once here and once in `trigger_new_era`.

Reply:
But these are disjointed paths, since we return at the end of this if block and do not call `trygger_new_era`. I will try to simplify the code.